docs/custom-gateway.md

```diff
@@ -37,22 +36,35 @@ User needs to implement following interface.
 
 ```go
 type ConfigExecutor interface {
-	StartConfig(configContext *ConfigContext) error
-	StopConfig(configContext *ConfigContext) error
+	StartConfig(configContext *ConfigContext)
+	StopConfig(configContext *ConfigContext)
+	Validate(configContext *ConfigContext) error
 }
 ```
 
-ConfigData contains configuration key/name and value.
+ConfigData contains configuration key/name, value and the state of the configuration 
 ```go
 type ConfigData struct {
-	// Src contains name of the configuration
-	Src    string
-	// Config contains the configuration
-	Config string
-	// StopCh is used to send a stop signal to configuration runner/executor
-	StopCh chan struct{}
+	// Data holds the actual configuration
+	Data *ConfigData
+
+	StartChan chan struct{}
+		
+	StopChan chan struct{}
+
+	DataChan chan []byte
+
+	DoneChan chan struct{}
+
+	ErrChan chan error
+
+	ShutdownChan chan struct{}
+
 	// Active tracks configuration state as running or stopped
 	Active bool
+	// Cancel is called to cancel the context used by client to communicate with gRPC server.
+	// Use it only if gateway is implemented as gRPC server.
+	Cancel context.CancelFunc
```

Comment:
this is cool but should it be restricted to use with `gRPC` server.. we should be using it for everything!